### DISCUSSION TITLE
Move away from `pathAndArgs` in @trpc/react

### DISCUSSION THREAD

**Discussion by colinhacks** [trpc/trpc#267] [Ideas]
Really not a fan of the `pathAndArgs` syntax. Is there a reason for it? Instead of `trpc.useQuery(path, input, opts)`? Is this a relic of when you were supporting multiple inputs?

**Comment by KATT** [comment]
The main reason is that the input arg is optional

**Comment by KATT** [comment]
I had it before that could do both useQuery
(path) for things that didn't have input and useQuery([path, input]) for things that did but that weirdly broke autocompletion on the path

**Comment by colinhacks** [comment]
The second argument should be conditional on whether Input is undefined. Pseudocode:

```ts
useQuery(path, args: Input extends undefined ? Opts : Input, opts?: Input extends undefined ? never : Opts);
```

I can submit a PR for this. So you can see how it looks.

**Comment by KATT** [comment]
The thing is, the input can also be _optional_, and that doesn't work nicely in your suggestion as you'd be forced to pass like undefined/null in case you want to supply options.

Id actually rather see if we can maybe mix in input in the query options and do like

useQuery(path, { input, ...otherOpts })
